src/MathOptFormat.jl

```diff
@@ -174,17 +197,7 @@ Create a MOI model by reading `filename`. Type of the returned model depends on
 the extension of `filename`.
 """
 function read_from_file(filename::String)
-    model = if endswith(filename, ".mof.json.gz") || endswith(filename, ".mof.json")
-        MOF.Model()
-    elseif endswith(filename, ".cbf.gz") || endswith(filename, ".cbf")
-        CBF.Model()
-    elseif endswith(filename, ".mps.gz") || endswith(filename, ".mps")
-        MPS.Model()
-    elseif endswith(filename, ".lp.gz") || endswith(filename, ".lp")
-        LP.Model()
-    else
-        error("File-type of $(filename) not supported by MathOptFormat.jl.")
-    end
+    model = _filename_to_model(filename)
```

Comment:
```Julia
function read_from_file(
    filename::String; 
    compression::AbstractCompressionScheme = AutomaticCompression(),
    file_format::FileFormat = AUTOMATIC_FILE_FORMAT,
)
    if file_format == AUTOMATIC_FILE_FORMAT
        for (format, (ext, _)) in _FILE_FORMATS
            if endswith(filename, ext) || occursin("$(ext).", filename)
                file_format = format
                break
            end
        end
        error(
            "Unable to detect automatically format of $(filename). Use the " *
            "`file_format` keyword to specify the file format."
        )
    end
    model = _FILE_FORMATS[file_format]()
    MOI.read_from_file(model, filename; compression = compression)
    return model
end
```

You might want to do something similar with `AutomaticCompression` so that we don't try `NoCompression` if the user passes in a weird filename.